app/notification_test.go

```diff
@@ -344,6 +344,42 @@ func TestGetExplicitMentions(t *testing.T) {
 				ChannelMentioned: true,
 			},
 		},
+		"Don't include potential mention that's part of an actual mention (without trailing period)": {
```

Comment:
Can you elaborate on the "Don't include potential mention" part of this? I'm a little unclear what that means, given that there's only one "potential mention" in all of these new test cases, and all of them get "included" by virtue of matching.

Reply:
I'll add a couple comments since I was really having trouble trying to explain myself in the name of the test. Basically, if we mention `@user.name` and that's an `@user` that exists outside the channel, we shouldn't assume someone is trying to mention `@user` since it's unlikely that they're trying to do that